tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/passes/VCGenPassImpl.scala

```diff
@@ -1,40 +1,41 @@
 package at.forsyte.apalache.tla.bmcmt.passes
 
 import at.forsyte.apalache.infra.passes.Pass.PassResult
-import at.forsyte.apalache.infra.passes.PassOptions
 import at.forsyte.apalache.tla.bmcmt.VCGenerator
 import at.forsyte.apalache.tla.lir.{ModuleProperty, TlaModule}
 import at.forsyte.apalache.io.lir.TlaWriterFactory
 import at.forsyte.apalache.tla.lir.transformations.TransformationTracker
 import com.google.inject.Inject
 import com.typesafe.scalalogging.LazyLogging
+import at.forsyte.apalache.infra.passes.DerivedPredicates
 
 /**
  * The pass that generates verification conditions.
  *
  * @author
  *   Igor Konnov
  */
-class VCGenPassImpl @Inject() (options: PassOptions, tracker: TransformationTracker, writerFactory: TlaWriterFactory)
+class VCGenPassImpl @Inject() (
+    derivedPreds: DerivedPredicates,
+    tracker: TransformationTracker,
+    writerFactory: TlaWriterFactory)
     extends VCGenPass with LazyLogging {
 
   override def name: String = "VCGen"
 
   override def execute(tlaModule: TlaModule): PassResult = {
     val newModule =
-      options.get[List[String]]("checker", "inv") match {
-        case Some(invariants) =>
+      derivedPreds.invariants match {
+        case List() =>
+          logger.info("  > No invariant given. Only deadlocks will be checked")
```

Comment:
Shouldn't this check for `--no-deadlocks`?

Reply:
Maybe! Perhaps I'll add that as a followup tho, as this just inherits the current logic.

Reply:
https://github.com/informalsystems/apalache/issues/2118